sources/osgUtil/Composer.js

```diff
@@ -164,51 +176,83 @@ define( [
                     }
                 }
 
-                // check if we want to render on screen
+
                 var camera = new Camera();
-                camera.setStateSet( element.filter.getStateSet() );
+                self._cameraRTT.push( camera );
+                camera.setStateSet( stateSet );
 
-                var texture;
-                var quad;
+
+                var textureResult;
+                // check if we want to render on screen
                 if ( lastFilterRenderToScreen === true ) {
                     w = self._renderToScreenWidth;
                     h = self._renderToScreenHeight;
                 } else {
                     camera.setRenderOrder( Camera.PRE_RENDER, 0 );
-                    texture = element.texture;
+                    textureResult = element.texture;
                     var textureTarget = element.textureTarget;
-                    if ( texture === undefined ) {
-                        texture = new Texture();
-                        texture.setTextureSize( w, h );
+                    if ( textureResult === undefined ) {
+                        textureResult = new Texture();
+                        textureResult.setName( 'composer Rtt ' + element.filter._fragmentName );
+                        textureResult.setTextureSize( w, h );
                         textureTarget = Texture.TEXTURE_2D;
+                        self._textureRTT.push( textureResult );
                     }
-                    camera.attachTexture( FrameBufferObject.COLOR_ATTACHMENT0, texture, textureTarget );
+                    camera.attachTexture( FrameBufferObject.COLOR_ATTACHMENT0, textureResult, textureTarget );
+
                 }
 
                 var vp = new Viewport( 0, 0, w, h );
                 camera.setReferenceFrame( TransformEnums.ABSOLUTE_RF );
                 camera.setViewport( vp );
+
+                // FIXME: not really useful, but osgjs keep pushing projection matrix
+                // and maybe some old code still use it
                 Matrix.makeOrtho( 0, 1, 0, 1, -5, 5, camera.getProjectionMatrix() );
 
-                quad = Shape.createTexturedFullScreenFakeQuadGeometry();
+                var quad = Shape.createTexturedFullScreenFakeQuadGeometry();
 
                 if ( element.filter.buildGeometry !== undefined )
                     quad = element.filter.buildGeometry( quad );
 
                 quad.setName( 'composer layer' );
 
-                lastTextureResult = texture;
+                // if rendering into texture framebuffer
+                if ( textureResult ) {
```

Comment:
I prefer you add function call for this instead of inline code, it's better to avoid spamming debug/devel code. If it was two line that would be ok, but if more it's better to keep this outside of core function. Maybe we should add this into the coding style